Try fix WP nightly unit tests

claudiosanches opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Recently WordPress removed [PHPUnit SpeedTrap Listener class](https://github.com/WordPress/wordpress-develop/commit/aae784273f922d7434ed2a0bf5e032ec7133bdc4#diff-bc27040b800d7215d4fd505e7f6c850790ace4f7d572a0b8d7626e6bc4d25acf) and implements `yoast/phpunit-polyfills` that's now a requirement to run Unit Tests using the WordPress test framework, see more details in this commit from WP core: https://github.com/WordPress/wordpress-develop/commit/9b565f7907ca1d4cb32eccc980de98683758df17

Note that Jetpack is also trying to solve it by using this new constant: https://github.com/Automattic/jetpack/pull/20618

### How to test the changes in this Pull Request:

Verify that all the unit test related checks pass in this pull request.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

claudiosanches committed:
Try fix WP nightly unit tests

claudiosanches committed:
Set PHPUNIT polyfills

claudiosanches committed:
Remove PHPUnit speedtrap

claudiosanches committed:
Merge branch 'trunk' into fix/unit-tests

Konamiman commented:
Hi @claudiosanches. Looks like the issue in the failing job is not caused by PHP 8 itself, but by the state of the autoloader after the custom PHPUnit is loaded in `ci.yml`. I've been able to reproduce the problem locally and it seems that after [the "Add PHP8 Compatibility" block](https://github.com/woocommerce/woocommerce/blob/fix/unit-tests/.github/workflows/ci.yml#L64-L68) runs there are two copies oh PHPUnit in place: one in `vendor/phpunit` and another one in `bin/composer/phpunit`; the autoloader is picking the former but it should pick the later instead.

So, locally, I've been able to make the tests work by running the following after the custom PHPUnit is installed:

```
rm -rf ./vendor/phpunit/
composer dump-autoload
```

Not sure why it stopped working, though; maybe due to a recent change in WordPress?

Konamiman committed:
Merge branch 'trunk' into fix/unit-tests

Konamiman committed:
Fix unit tests failing in PHP 8 in CI

The problem was that PHP 8 requires setting up a different version of
PHPUnit, and after doing that two versions were in place, one in
'vendor/phpunit' and another one in 'bin/composer/phpunit'; the
autoloader was picking the former but it should pick the later.

roykho commented on phpunit.xml:
```diff
@@ -46,7 +46,4 @@
 		</whitelist>
 	</filter>
-	<listeners>
-		<listener class="SpeedTrapListener" file="tests/legacy/includes/listener-loader.php" />
```
If we're no longer using this, perhaps we could remove this file as well?

Konamiman replied:
This file is still included in the code files inside `test/legacy`. So either we can remove that folder as well, or we need to keep the file.

roykho replied:
Can you show me where? Because when I did a search, the only place `listener-loader.php` shows up is the `listener-loader.php` file itself and in the `phpunit.xml` which you removed.

Konamiman replied:
Of course. WP's `listener-loader.php`  is referenced in Woo's `listener-loader.php` and that's it... me being supersmart as usual. The file is now removed.

Konamiman committed:
Remove listener-loader.php as it's no longer used.

roykho reviewed (approved):
LGTM!
